apps/ui/pages/[schema]/pages/index.vue

```diff
@@ -0,0 +1,418 @@
+
+  $fetch(`/${schema}/graphql`, {
+    method: "POST",
+    body: {
+    query: `mutation drop($settings:[DropSettingsInput]){drop(settings:$settings){status message}}`,
+    variables: {
+      settings: {
+        key: `page.${pageToDelete.value}`,
+      }
+    }
+    }
+  })
+  .then((response) => {
+    if (response?.errors) {
+      throw new Error(`Unable to delete page '${pageToDelete.value}': ${response.errors[0].message}`);
+    }
+
+    statusModalData.value = {
+      type: "success",
+      message:`Deleted '${pageToDelete.value}'`
+    };
+
+    fetchPages();
+  })
+  .catch((err) => {
+    statusModalData.value = { type: "error", message: err };
+  })
+  .finally(() => {
+    showDeleteModal.value = false;
+    showStatusModal.value = true;
+  });
+}
+
+function handleSort($event: Record<string,string>) {
+  Object.keys(metadata.value)
+    .forEach((key: string) => metadata.value[key] = { isSorted: false });
+  metadata.value[$event.column as string] = { isSorted: true }
+  sortData($event.column as string, $event.sort === 'ASC')
+}
+
+function sortData(column: string, isAscending: boolean) {
+  pages.value = pages.value.sort((a: Page,b: Page) => {
+    const aValue = a[column as keyof Page] as string;
+    const bValue = b[column as keyof Page] as string;
+    if (isAscending) {
+     if (aValue === "" || aValue === undefined) {
+      return 1;
+     } else if (bValue === "" || bValue === undefined) {
+      return -1;
+     } else {
+      return aValue?.localeCompare(bValue);
+     }
+    } else {
+      if (aValue === "" || aValue === undefined) {
+       return -1;
+      } else if (bValue === "" || bValue === undefined) {
+       return 1;
+      } else {
+       return bValue?.localeCompare(aValue);
+      }
+    }
+  });
+}
+
+const crumbs: Record<string, string> = {};
+if (schema) {
+  crumbs[schema] = `/${schema}`;
+}
+crumbs["Pages"] = "";
+</script>
+
+<template>
+  <div class="mx-auto px-0 lg:px-7.5">
+    <PageHeader title="Pages" align="left">
+      <template #prefix>
+        <BreadCrumbs align="left" :crumbs="crumbs" />
+      </template>
+    </PageHeader>
+    <div class="flex justify-end gap-2.5 mb-7.5" v-if="isAdmin">
+      <Button type="primary" icon="add-circle" @click="showAddPageModal = true">
+        Add page
+      </Button>
+      <Modal
+        title="Add page"
+        v-model:visible="showAddPageModal"
+        max-width="max-w-9/10"
+      >
+        <div title="Add page" class="text-title-contrast p-7.5 overflow-y-auto">
+          <form @click="$event.stopPropagation()" @submit.prevent>
+            <legend class="sr-only">Add a new page</legend>
+            <label class="font-bold">
+              Name
+            </label>
+            <p>Name cannot contain and numbers, characters, or spaces</p>
+            <InputString
+              id="manage-pages-new-page-name"
+              placeholder="MyNewPage"
+              class="mt-2"
+              v-model="newPageName"
```

Comment:
I didn't get an error when having faulty name.
Should be invalid with error message right?
Also I would prefer if I could use '-' in my slug, or is that also non-standard?